crates/cacache/src/put.rs

```diff
@@ -179,15 +179,16 @@ where
     D: AsRef<[u8]>,
     K: AsRef<str>,
 {
-    let mut writer = SyncWriter::create(cache.as_ref(), key.as_ref())?;
-    writer.write_all(data.as_ref()).with_context(|| {
+    let data = data.as_ref();
+    let mut writer = SyncWriter::create_with_size(cache.as_ref(), key.as_ref(), data.len())?;
```

Comment:
This forwards the size of the incoming data down to the Writer so we don't have to backtrack to write it when the stream finishes.